### PR TITLE
Return a count of the number of elements visited.

### LINKED ISSUE #69: QueueFile#forEach(ElementVisitor reader) should return number of elements read
Currently `QueueFile#forEach(ElementVisitor reader)` returns void, which means callers have to keep track of how many elements were read.

```java
int read = queueFile.forEach(visitor);
queueFile.remove(read);
```

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #69.

### CODE REVIEW

File: tape/src/main/java/com/squareup/tape/QueueFile.java [hunk 1]
```diff
@@ -440,15 +440,20 @@ public synchronized void peek(ElementVisitor visitor) throws IOException {
    * Invokes the given reader once for each element in the queue, from eldest to most recently
    * added. Continues until all elements are read or {@link ElementVisitor#read reader.read()}
    * returns {@code false}.
+   *
+   * @return number of elements visited
    */
-  public synchronized void forEach(ElementVisitor reader) throws IOException {
+  public synchronized int forEach(ElementVisitor reader) throws IOException {
     int position = first.position;
     for (int i = 0; i < elementCount; i++) {
```
Comment (f2prateek) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: FWIW, I did `for (int i = 1; i <= elementCount; i++) ` and returned `i` below.

**Comment by f2prateek** [comment]
:shipit:

File: tape/src/main/java/com/squareup/tape/QueueFile.java [hunk 2]
```diff
@@ -440,15 +440,20 @@ public synchronized void peek(ElementVisitor visitor) throws IOException {
    * Invokes the given reader once for each element in the queue, from eldest to most recently
    * added. Continues until all elements are read or {@link ElementVisitor#read reader.read()}
    * returns {@code false}.
+   *
+   * @return number of elements visited
    */
-  public synchronized void forEach(ElementVisitor reader) throws IOException {
+  public synchronized int forEach(ElementVisitor reader) throws IOException {
```
Comment (swankjesse) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: FYI, not binary compatible

Comment (JakeWharton) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: We haven't shipped this method yet!

File: tape/src/main/java/com/squareup/tape/QueueFile.java [hunk 3]
```diff
@@ -440,15 +440,20 @@ public synchronized void peek(ElementVisitor visitor) throws IOException {
...
     int position = first.position;
     for (int i = 0; i < elementCount; i++) {
       Element current = readElement(position);
```
Comment (swankjesse) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: What if this is Element.NULL ?

File: tape/src/test/java/com/squareup/tape/QueueFileTest.java [hunk 4]
```diff
@@ -548,7 +548,7 @@ public void testForEachReadWithOffset() throws IOException {
     queueFile.add(b);
 
     final int[] iteration = new int[] { 0 };
```
Comment (swankjesse) [inline tape/src/test/java/com/squareup/tape/QueueFileTest.java]: `AtomicInteger` ?

File: tape/src/main/java/com/squareup/tape/QueueFile.java (hunk 3 above)
Comment (swankjesse) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: ahhhh, elementCount catches that.

File: tape/src/test/java/com/squareup/tape/QueueFileTest.java (hunk 4 above)
Comment (JakeWharton) [inline tape/src/test/java/com/squareup/tape/QueueFileTest.java]: Don't look at the tests. They're filthy... spring cleaning them soon.

**Pull request merged**